src/rpc/mining.cpp

```diff
@@ -543,7 +544,10 @@ UniValue getblocktemplate(const JSONRPCRequest& request)
 
         // Create new block
         CScript scriptDummy = CScript() << OP_TRUE;
-        pblocktemplate = BlockAssembler(Params()).CreateNewBlock(scriptDummy, fSupportsSegwit);
+	
+	//TODO: Come back and review how to get drivechain commitments here
+        std::vector<CTxOut> drivechainCommitments;
```

Comment:
Your idea is basically the same as what I've come up with, I'll talk to @psztorc about it tomorrow as well and get back to you :)